src/btree/bt_curnext.c

```diff
@@ -188,15 +188,16 @@ __cursor_var_append_next(
 
         if (cbt->upd_value->type == WT_UPDATE_INVALID) {
             ++*skippedp;
+            cbt->non_deleted_updates = true;
             continue;
         }
         if (cbt->upd_value->type == WT_UPDATE_TOMBSTONE) {
-            if (cbt->upd_value->tw.stop_txn != WT_TXN_NONE &&
-              __wt_txn_upd_value_visible_all(session, cbt->upd_value))
-                ++cbt->page_deleted_count;
+            __wt_cbt_set_non_deleted_updates_flag(session, cbt, cbt->ins->upd);
             ++*skippedp;
             continue;
         }
+
+        cbt->non_deleted_updates = true;
```

Comment:
Should we set this flag based on the isolation of the session? If a reader has read committed isolation, I think this check is wrong.

Reply:
Irrespective of isolation level, if all the keys on a page are removed, it is better to evict that page, so that the eligible transactions can skip traversing this page as part of the tree walk. Let me know if you think of any scenario where it can cause problems?

Reply:
If we are in read uncommitted isolation and all the keys are removed by an uncommitted transaction, we will evict that as well? If so, we will waste of our work because the page cannot be evicted.

Reply:
Understood. We can go wrong if we have an uncommitted transaction that performs the delete/truncate operation. I added a check where we set the page to be evict soon for the transaction isolation check.